---
output: 
  html_document: 
    css: styles_20180224.css
    df_print: kable
    highlight: tango
    keep_md: yes
    number_sections: yes
    self_contained: no
    smart: no
    theme: readable
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  fig.path = "images_20180224",
  cache=TRUE,
  background = "#FFFFFF",
  fig.align='center'
)
```


The ReCounts project <http://bowtie-bio.sourceforge.net/recount/> is a Johns Hopkins University project that has made the transcript counts of processed RNA-Seq data from multiple published papers publicly available. My objective in the current project is to download and analyze the count data for one paper in particular: *"Transcript assembly and quantification by RNA-Seq reveals unannotated transcripts and isoform switching during cell differentiation."* by [Trapnell et. al., 2010](https://www.ncbi.nlm.nih.gov/pubmed?term=20436464).


The Trapnell paper performed RNA-Seq on the cell line C2C12 from C57BL/6 mice at 4 time points during a time course experiment. Cells were cultured in 20% fetal bovine serum in DMEM until confluent and then switched to low serum medium for 168 hours. The time points sampled were -24, 60, 120, and 168 hours after switching to low serum medium.


The analysis that follows is broken into several key steps:

* Define key packages
* Data acquisition
* Data loading
* Data wrangling
* Diffusion mapping
* Differential expression analysis
* Conclusion


# Define key packages
```{r,warning=FALSE,results='hide',message=FALSE}
library(readr)
library(magrittr)
library(tidyr)
library(dplyr)
library(ggplot2)

# source("https://bioconductor.org/biocLite.R")
# biocLite("destiny")
library(destiny)
# biocLite("DESeq2")
library(DESeq2)
```


# Data acquisition
```{r,cache=TRUE,message=FALSE,results='hide'}
# Define data URLs.
count_url <- "http://bowtie-bio.sourceforge.net/recount/countTables/trapnell_count_table.txt"
phenotype_url <- "http://bowtie-bio.sourceforge.net/recount/phenotypeTables/trapnell_phenodata.txt"
gene_info_url <- "http://bowtie-bio.sourceforge.net/recount/ivals/mouse_genes.txt"

# Define file names.
count_file <- "counts.txt"
pheno_file <- "phenotypes.txt"
gene_info_file <- "gene_info.txt"

# Download data files.
download.file(count_url, count_file)
download.file(phenotype_url, pheno_file)
download.file(gene_info_url, gene_info_file)
```


# Data loading
```{r,message=FALSE}
# Load data.
counts <- readr::read_delim(count_file, delim = "\t")
phenos <- readr::read_delim(pheno_file, delim = " ")
gene_info <- readr::read_delim(gene_info_file, delim = "\t")
```


# Data Wrangling
Here, the data is rearranged for subsequent analyses.
```{r}
# Create a wide count matrix for diffusion mapping (Sample x Gene).
count_mat <- counts %>%
  as.data.frame() %>%
  tibble::column_to_rownames(., "gene") %>%
  as.matrix() %>%
  t()

# Create a stacked (tall) count tibble and add pheno data.
count_tib <- counts %>%
  tidyr::gather(., key = "Sample", value = "Count", -gene)
time_dict <- phenos %>%
  dplyr::select(., sample.id, time.point.hrs) %>%
  tibble::deframe()
count_tib <- count_tib %>%
  dplyr::mutate(., Time = time_dict[Sample])
```


# Diffusion mapping
Diffusion mapping or differentiation mapping is a Markov Chain-based method that allows the transition probabilities from different cell states to be identified. I used the diffusion mapping implementation in the [*destiny* package](https://academic.oup.com/bioinformatics/article/31/18/2989/241305).


## Perform diffusion mapping
```{r}
# Get the diffusion components for each sample.
dfm_object <- count_mat %>%
  destiny::DiffusionMap() %>%
  destiny::as.data.frame() %>%
  tibble::as_tibble() %>%
  dplyr::mutate(
    ., 
    Sample = rownames(count_mat),
    Time = factor(time_dict[Sample])
  )
```


## Plot diffusion mapping result
With a sample size of 4, the results aren't meaningful. I will acquire a larger sample size and analyze it in a future project.
```{r}
qplot(DC1, DC2, data = dfm_object, colour = Time)+
  theme_bw()
```


# Differential expression analysis
## Create the DESeqDataSet object
```{r,message=FALSE}
dds <- DESeq2::DESeqDataSetFromMatrix(
  countData = as.data.frame(counts) %>% tibble::column_to_rownames("gene"),
  colData = DataFrame(
    dplyr::mutate(
      phenos, Time = factor(time.point.hrs)
    )
  ),
  design = ~ Time
)
```


## Analyze expression levels
DESeq2 normalizes counts data by dividing the counts for each gene by the geometric mean of the corresponding gene counts as described [here](https://chipster.csc.fi/manual/deseq2.html).
```{r,message=FALSE,warning=FALSE}
dds <- DESeq2::DESeq(dds)
```


## Extract transformed data
Two of the main transformation methods are the variance stabilizing transformation and the regularized log transformation.

* [**Variance stabilizing transformation.**](https://en.wikipedia.org/wiki/Variance-stabilizing_transformation) The variance in expression levels is typically positively related to the expression level itself. To facilitate 1) graphical analyses and 2) the application of an ANOVA, the data is transformed in such a manner as to put the values for all genes on roughly the same scale. Since the relationship between the mean and variance of gene expression data often resembles a Poisson distribution, an [Anscome transformation](https://en.wikipedia.org/wiki/Anscombe_transform) is often appropriate. This method requires technical replicates to estimate the dispersion and thus is not appropriate here.
* [**Regularized log transformation.**](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8) This method addresses the same problem as the variance stabilizing transformation, however, does so by taking the base 2 logarithm of the normalized counts data plus 0.5.
```{r,message=FALSE}
# Perform the regularized log transformation.
rld <- DESeq2::rlog(dds, blind = FALSE)
```


## Heatmap of sample distances
```{r,message=FALSE}
# Create a matrix of sample distances.
sample_distances <- rld %>%
  SummarizedExperiment::assay() %>%
  t() %>%
  dist() %>%
  as.matrix()
rownames(sample_distances) <- phenos$time.point.hrs
colnames(sample_distances) <- phenos$time.point.hrs

# Plot the sample distances matrix.
gplots::heatmap.2(
  sample_distances,
  key = FALSE, trace = "none"
)
```


## Get differential expression results
```{r}
# Calculate differential expression results.
diff_expr <- DESeq2::results(dds)
```


## Plot adjusted p-values
```{r,warning=FALSE,message=FALSE}
# Create a dictionary to convert from ensemble_gene_id to gene_id.
gene_dict <- gene_info %>%
  dplyr::select(., ensembl_gene_id, external_gene_id) %>%
  tibble::deframe()

# Extract differential expression data.
de_df <- tibble::as_tibble(diff_expr) %>%
  dplyr::mutate(
    .,
    ensemble_gene_id = rownames(diff_expr),
    Gene = gene_dict[ensemble_gene_id]
  )

# Plot the adjusted p-values as a histogram.
qplot(x = padj, data = de_df)+theme_bw()+
  scale_x_continuous(expand = c(0, 0))+
  scale_y_continuous(expand = c(0, 0))
```


## Identify differentially expressed genes
```{r,warning=FALSE,message=FALSE}
de_genes <- de_df %>%
  dplyr::filter(., padj < 0.05) %>%
  dplyr::arrange(., padj, abs(log2FoldChange))

de_genes %>%
  dplyr::select(., Gene, log2FoldChange, padj)
```


## Summary of the differential expression results
`r nrow(de_genes)` differentially expressed genes were identified at a significance level of 0.05. Of these genes, `r sum(de_genes$log2FoldChange > 0)` were upregulated and `r sum(de_genes$log2FoldChange < 0)` were downregulated. These expression changes are depicted below.
```{r}
qplot(Gene, log2FoldChange, data = de_genes)+
  theme_bw()+
  scale_y_continuous(expand = c(0, 0))+
  theme(
    axis.text.x = element_text(angle = 90)
  )
```


# Conclusion
Since the downloaded data is limited to 4 samples (just 1 per time point), the ability to identify differentially expressed genes is quite limited. Nevertheless several interesting differentially expressed genes were identified, including myoglobin (Mb) and Apobec2, which is an mRNA editing enzyme. In addition, performing diffusion mapping on such a small sample size will almost always fail to reveal reliable results.


# Next steps
To overcome the limitations of small sample sizes, single-cell RNA-Seq data could be used. This will be the focus of a future analysis.
